### PR TITLE
fix: data-uri icons in edit visible assets modal

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/32676

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See STR in issue

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/asset-watchlist-item/index.tsx [hunk 1]
```diff
@@ -93,7 +92,7 @@ const AssetWatchlistItem = React.forwardRef<HTMLDivElement, Props>(
     return (
       <StyledWrapper ref={forwardedRef}>
         <NameAndIcon onClick={onClickAsset}>
-          {token.isErc721 && !isDataURL(token.logo) ? (
+          {token.isNft ? (
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/asset-watchlist-item/index.tsx:95]: I thought `isNft` was reserved for Solana (please correct me if not the case).

If the above is true, a more comprehensive condition would be:

```ts
token.isNft || token.isErc721 || token.isErc1155
```

Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/asset-watchlist-item/index.tsx:95]: `isNFT` Works for everything: https://github.com/brave/brave-browser/issues/26395

Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/asset-watchlist-item/index.tsx:95]: [Auto detect token standard when adding NFT assets](https://github.com/brave/brave-core/pull/17207#top)

File: components/brave_wallet_ui/components/desktop/portfolio-asset-item/index.tsx [hunk 2]
```diff
@@ -104,7 +103,7 @@ export const PortfolioAssetItem = ({
   )
 
   // memos & computed
-  const isNonFungibleToken = React.useMemo(() => token.isNft || token.isErc721, [token.isNft, token.isErc721])
+  const isNonFungibleToken = token.isNft || token.isErc721
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/portfolio-asset-item/index.tsx:106]: `token.isErc721` not required, following your previous comment?

**Review by onyb — APPROVED** [review]
++

left a small comment if you want to address it

**Pull request merged**